e2e/framework/util/util.go

```diff
@@ -276,32 +279,52 @@ func GetKubeSecret(client kubernetes.Interface, namespace, secretName string) (*
 }
 
 // NewConfig loads and returns the kubernetes credentials from the environment.
-// KUBECONFIG env var takes precedence and falls back to in-cluster config.
+// KUBECONFIG env var takes precedence, falls back to in-cluster config, then to default KUBECONFIG location.
 func NewConfig() (*restclient.Config, *kubernetes.Clientset, crclient.Client) {
-	var kubeConfig *restclient.Config
-	var err error
-	kcPath := os.Getenv("KUBECONFIG")
-	if kcPath != "" {
-		kubeConfig, err = clientcmd.BuildConfigFromFlags("", kcPath)
-		if err != nil {
-			Fail(err.Error())
-		}
-	} else {
-		kubeConfig, err = restclient.InClusterConfig()
-		if err != nil {
-			Fail(err.Error())
-		}
+	cfg, err := BuildKubeConfig()
+	if err != nil {
+		Fail(err.Error())
```

Comment:
why do you not use expectations here?

Reply:
I guess there are two ways to handle this: 
1.use gomega expectation `Expect(err).ToNot(HaveOccurred())`  + `defer GinkgoRecover()` - because the `Expect` panics - and this has to be recovered, otherwise Ginkgo is unhappy and yells at you that you shouldn't do this. The panic is a problem, because it is called from a ginkgo _container_ node, not a _subject_ node (which handles panics gracefully).
2. or option B: use `Fail()` - which also panics but ginkgo is aware of it and does its thing to fail the test subject, print errors  etc.

To me, the `defer` approach seems more convoluted. Though both ways are shitty and inconsistent :cry:

Reply:
agree on that. I guess im wondering why we even panic in the first place?